projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -0,0 +1,531 @@
+    * @example
+    *```html
+    * <input igxDateTimeEditor [minValue]="minDate">
+    *```
+    */
+  @Input()
+  public minValue: string | Date;
+
+  /**
+  * An @Input property that allows you to set the maximum possible value the editor will allow.
+  * @example
+  *```html
+  * <input igxDateTimeEditor [maxValue]="maxDate">
+  *```
+  */
+  @Input()
+  public maxValue: string | Date;
+
+  /**
+   * An @Input property that allows you to specify if the currently spun date segment should loop over.
+   * @example
+   *```html
+   * <input igxDateTimeEditor [isSpinLoop]="false">
+   *```
+   */
+  @Input()
+  public isSpinLoop = true;
+
+  /**
+   * An @Input property that allows you to set both pre-defined format options such as `shortDate` and `longDate`,
+   * as well as constructed format string using characters supported by `DatePipe`, e.g. `EE/MM/yyyy`.
+   * @example
+   *```html
+   * <input igxDateTimeEditor [displayFormat]="'shortDate'">
+   *```
+   */
+  @Input()
+  public displayFormat: string;
+
+  /**
+   * An @Input property that allows you to get/set the expected user input format(and placeholder).
+   *  for the editor.
+   * @example
+   *```html
+   * <input [igxDateTimeEditor]="'dd/MM/yyyy'">
+   *```
+   */
+  @Input(`igxDateTimeEditor`)
+  public set inputFormat(value: string) {
+    if (value) {
+      this._format = value;
+    }
+    const mask = this.buildMask(this.inputFormat);
+    this.mask = value.indexOf('tt') !== -1 ? mask.substring(0, mask.length - 2) + 'LL' : mask;
+  }
+
+  public get inputFormat(): string {
+    return this._format;
+  }
+
+  /**
+   * An @Input property that gets/sets the component date value.
+   * @example
+   * ```html
+   * <input igxDateTimeEditor [value]="date">
+   * ```
+   */
+  @Input()
+  public set value(value: Date) {
+    this._value = value;
+    this.updateMask();
+  }
+
+  public get value() {
+    return this._value;
+  }
+
+  /**
+   * Emitted when the editor's value has changed.
+   * @example
+   * ```html
+   * <input igxDateTimeEditor (valueChanged)="onValueChanged($event)"/>
+   * ```
+   */
+  @Output()
+  public valueChanged = new EventEmitter<IgxDateTimeEditorEventArgs>();
+
+  /**
+   * Emitted when the editor is not within a specified range.
+   * @example
+   * ```html
+   * <input igxDateTimeEditor [minValue]="minDate" [maxValue]="maxDate" (validationFailed)="onValidationFailed($event)"/>
+   * ```
+   */
+  @Output()
+  public validationFailed = new EventEmitter<IgxDateTimeEditorEventArgs>();
+
+  private _value: Date;
+  private _document: Document;
+  private _isFocused: boolean;
+  private _format = 'dd/MM/yyyy';
+  private _oldValue: Date | string;
+  private _dateTimeFormatParts: DatePartInfo[];
+  private onTouchCallback = (...args: any[]) => { };
+  private onChangeCallback = (...args: any[]) => { };
+
+  private get literals() {
+    const literals = [];
+    for (const char of this.mask) {
+      if (char.match(/[^0lL]/)) { literals.push(char); }
```

Comment:
What's the L deal? A literal for a date mask **can** be in any other letter that's not in the original `tt` bit, right? This can create cause `AM` as part of a string to be added to the literals here.
Do rely ONLY on the date parts array, for everything and drop other logic that tries to do "mask stuff"